nistats/tests/test_first_level_model.py

```diff
@@ -0,0 +1,308 @@
+    labels, results = run_glm(Y, X, 'ols')
+    assert_array_equal(labels, np.zeros(n))
+    assert_equal(list(results.keys()), [0.0])
+    assert_equal(results[0.0].theta.shape, (q, n))
+    assert_almost_equal(results[0.0].theta.mean(), 0, 1)
+    assert_almost_equal(results[0.0].theta.var(), 1. / p, 1)
+
+    # ar(1) case
+    labels, results = run_glm(Y, X, 'ar1')
+    assert_equal(len(labels), n)
+    assert_true(len(results.keys()) > 1)
+    tmp = sum([val.theta.shape[1] for val in results.values()])
+    assert_equal(tmp, n)
+
+    # non-existant case
+    assert_raises(ValueError, run_glm, Y, X, 'ar2')
+    assert_raises(ValueError, run_glm, Y, X.T)
+
+
+def test_scaling():
+    """Test the scaling function"""
+    shape = (400, 10)
+    u = np.random.randn(*shape)
+    mean = 100 * np.random.rand(shape[1]) + 1
+    Y = u + mean
+    Y_, mean_ = mean_scaling(Y)
+    assert_almost_equal(Y_.mean(0), 0, 5)
+    assert_almost_equal(mean_, mean, 0)
+    assert_true(Y.std() > 1)
+
+
+def test_fmri_inputs():
+    # Test processing of FMRI inputs
+    with InTemporaryDirectory():
+        shapes = ((7, 8, 9, 10),)
+        mask, FUNCFILE, _ = write_fake_fmri_data(shapes)
+        FUNCFILE = FUNCFILE[0]
+        func_img = load(FUNCFILE)
+        T = func_img.shape[-1]
+        conf = pd.DataFrame([0, 0])
+        des = pd.DataFrame(np.ones((T, 1)), columns=[''])
+        des_fname = 'design.csv'
+        des.to_csv(des_fname)
+        for fi in func_img, FUNCFILE:
+            for d in des, des_fname:
+                FirstLevelModel().fit(fi, design_matrices=d)
+                FirstLevelModel(mask=None).fit([fi], design_matrices=d)
+                FirstLevelModel(mask=mask).fit(fi, design_matrices=[d])
+                FirstLevelModel(mask=mask).fit([fi], design_matrices=[d])
+                FirstLevelModel(mask=mask).fit([fi, fi], design_matrices=[d, d])
+                FirstLevelModel(mask=None).fit((fi, fi), design_matrices=(d, d))
+                assert_raises(
+                    ValueError, FirstLevelModel(mask=None).fit, [fi, fi], d)
+                assert_raises(
+                    ValueError, FirstLevelModel(mask=None).fit, fi, [d, d])
+                # At least paradigms or design have to be given
+                assert_raises(
+                    ValueError, FirstLevelModel(mask=None).fit, fi)
+                # If paradigms are given then both tr and slice time ref were
+                # required
+                assert_raises(
+                    ValueError, FirstLevelModel(mask=None).fit, fi, d)
+                assert_raises(
+                    ValueError, FirstLevelModel(mask=None, t_r=1.0).fit, fi, d)
+                assert_raises(
+                    ValueError, FirstLevelModel(mask=None, slice_time_ref=0.).fit, fi, d)
+            # confounds rows do not match n_scans
+            assert_raises(
+                ValueError, FirstLevelModel(mask=None).fit, fi, d, conf)
+
+
+def basic_paradigm():
```

Comment:
This test should not be here but in test_dmtx.py or test_experimental_paradigm.py